core/src/main/kotlin/fr/sncf/osrd/conflicts/Conflicts.kt

```diff
@@ -58,9 +151,9 @@ fun conflictDetectorFromRequirements(requirements: List<Requirements>): Conflict
 
 class ConflictDetectorImpl(requirements: List<Requirements>) : ConflictDetector {
     private val spacingZoneRequirements =
-        mutableMapOf<String, MutableList<SpacingZoneRequirement>>()
+        mutableMapOf<ZoneId, MutableList<SpacingZoneRequirement>>()
     private val routingZoneRequirements =
-        mutableMapOf<String, MutableList<RoutingZoneRequirement>>()
+        mutableMapOf<ZoneId, MutableList<RoutingZoneRequirement>>()
```

Comment:
We could look into using `IdxMap` here (ultimately using an `ArrayList` with one element per zone in the infra). But we'd need to run benchmarks to make sure it's actually better.

Reply:
For a normal study, I'd expect the map to be sparse, so this is better. For stdcm though, most zones probably have requirements, so an array is *probably* better...

Reply:
I'll leave it as it is for now, and test it in a different PR with benchmarks